src/cpp/core/Base64.cpp

```diff
@@ -13,51 +13,121 @@
  *
  */
 
-#include <sstream>
-
-#include <algorithm>
-
-#include <boost/archive/iterators/base64_from_binary.hpp>
-#include <boost/archive/iterators/binary_from_base64.hpp>
-#include <boost/archive/iterators/transform_width.hpp>
-#include <boost/archive/iterators/ostream_iterator.hpp>
-
+#include <core/Macros.hpp>
 #include <core/Error.hpp>
 #include <core/Log.hpp>
 #include <core/FileSerializer.hpp>
 
+#include <boost/scoped_array.hpp>
+
 namespace rstudio {
 namespace core {
 namespace base64 {
 
-Error encode(const std::string& input, std::string* pOutput)
+typedef unsigned char Byte;
+
+namespace {
+
+std::string table()
 {
-   using namespace boost::archive::iterators;
+   return "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
```

Comment:
What's our confidence level that this table is identical to the one used by Boost's encoder/decoder? Seems like we don't need to worry too much about compatibility since we don't use this implementation for crypto, just image encoding in htmlutils--sound right?